Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/RdbmsProviderTest.cs

```diff
@@ -70,11 +70,56 @@ public override void SetUp ()
 
       _provider = new TestableRdbmsProvider(
           TestDomainStorageProviderDefinition,
+          TestDomainStorageProviderDefinition.ConnectionString,
           NullPersistenceExtension.Instance,
           _commandFactoryMock.Object,
           () => _connectionCreatorMock.Object.CreateConnection());
     }
 
+
+    [Test]
+    public void RdbmsProvider_WithConnectionStringContainedInProviderDefinition ()
+    {
+      var rdbmsProvider = new RdbmsProvider(
+          new RdbmsProviderDefinition("dummyProviderDefinition", Mock.Of<IRdbmsStorageObjectFactory>(), "dummyConnectionString", "dummyReadOnlyConnectionString"),
+          "dummyConnectionString",
+          NullPersistenceExtension.Instance,
+          _commandFactoryMock.Object,
+          () => _connectionCreatorMock.Object.CreateConnection());
+
+      Assert.That(rdbmsProvider, Is.InstanceOf<RdbmsProvider>());
+    }
+
+    [Test]
+    public void RdbmsProvider_WithReadOnlyConnectionStringContainedInProviderDefinition ()
+    {
+      var rdbmsProvider = new RdbmsProvider(
+          new RdbmsProviderDefinition("dummyProviderDefinition", Mock.Of<IRdbmsStorageObjectFactory>(), "dummyConnectionString", "dummyReadOnlyConnectionString"),
+          "dummyReadOnlyConnectionString",
+          NullPersistenceExtension.Instance,
+          _commandFactoryMock.Object,
+          () => _connectionCreatorMock.Object.CreateConnection());
+
+      Assert.That(rdbmsProvider, Is.InstanceOf<RdbmsProvider>());
+    }
+
+    [Test]
+    public void RdbmsProvider_WithConnectionStringNotInProviderDefinition_ThrowsArgumentException ()
```

Comment:
I think we are missing the readonly-version for this excption.

Reply:
This test only throws if the supplied connection string is neither the normal nor the readonly connection string, so we should only have a single test here.

Reply:
@Frazerus ah, yes, you're correct.